lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,427 @@
+        self.percentiles_cube = percentiles_cube
+
+        if output_name is not None:
+            self.output_name = output_name
+        else:
+            self.output_name = "probability_of_{}".format(
+                percentiles_cube.name())
+
+        # Set inverse_ordering switch
+        percentile_slices = percentiles_cube.slices_over(
+            self.percentile_coordinate)
+        self.inverse_ordering = False
+        first_percentile = percentile_slices.next().data
+        for percentile_values in percentile_slices:
+            last_percentile = percentile_values.data
+        if (first_percentile - last_percentile >= 0).all():
+            self.inverse_ordering = True
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ProbabilitiesFromPercentiles2D: percentiles_cube: {}, '
+                  'output_name: {}, inverse_ordering: {}'.format(
+                      self.percentiles_cube, self.output_name,
+                      self.inverse_ordering))
+        return result
+
+    def create_probability_cube(self, cube):
+        """
+        Create a 2-dimensional probability cube in which to store the
+        calculated probabilities.
+
+        Args:
+            cube (iris.cube.Cube):
+                Template for the output probability cube.
+        Returns:
+            probability_cube (iris.cube.Cube):
+                A new 2-dimensional probability cube with suitable metadata.
+        """
+        cube_format = next(cube.slices([cube.coord(axis='y'),
+                                        cube.coord(axis='x')]))
+        probabilities = cube_format.copy(data=np.full(cube_format.shape,
+                                                      np.nan, dtype=float))
+        try:
+            probabilities.remove_coord(self.percentile_coordinate)
+        except CoordinateNotFoundError:
+            pass
+
+        probabilities.units = 1
+        probabilities.rename(self.output_name)
+        probabilities.attributes['relative_to_threshold'] = 'below'
+        if self.inverse_ordering is True:
+            probabilities.attributes['relative_to_threshold'] = 'above'
+        return probabilities
+
+    def percentile_interpolation(self, threshold_cube, percentiles_cube):
+        """
+        Using a percentiles_cube containing a distinct percentile distribution
+        for each point on a 2-dimensional grid, we can interpolate through each
+        distribution to obtain a probability. The point to which we interpolate
+        is defined by the threshold_cube.
+
+        Note that the current implementation assumes that in cases of a
+        degenerate percentile distribution, the right most bin in which a
+        threshold value is found is chosen.
+
+        e.g.::
+
+            Percentile: 0 10 20 30 40 50 ...
+            Height (m): 0 0 0 15 30 40 ...
+
+            A height of 0m will be associated with a probabilty of 20%. This is
```

Comment:
I don't think you want this paragraph to be rendered as code as well, it probably need to go out a level.

Reply:
✅

Reply:
This still looks weird in the documentation